core-derive/src/serialization.rs

```diff
@@ -154,3 +171,78 @@ fn determine_enum_ser_method(data: &DataEnum) -> Ident {
         _ => panic!("Unsupported enum value. Expected string or int literal."),
     }
 }
+
+// ============================================================================
+// Union Serialization
+// ============================================================================
+
+/// Generates body of serialization impl for Enums
+fn serialize_union(shape_name: &Ident, schema_ident: &Ident, data: &DataEnum) -> TokenStream {
```

Comment:
Just a general comment about methods in core-derive - it may help with readability to have a prefix to these methods, i.e. `gen_serialize_union` or `impl_serialize_union`. 

Not something we need to address right now, we can change later as we add more and more in this crate.